classes/search/ArticleSearchIndex.inc.php

```diff
@@ -277,11 +271,11 @@ public function rebuildIndex($log = false, $journal = null, $switches = array())
 
 				if ($log) echo __('search.cli.rebuildIndex.indexing', array('journalName' => $journal->getLocalizedName())) . ' ... ';
 
-				$articles = $submissionDao->getByContextId($journal->getId());
-				while ($article = $articles->next()) {
-					if ($article->getSubmissionProgress() == 0) { // Not incomplete
-						$this->submissionMetadataChanged($article);
-						$this->submissionFilesChanged($article);
+				$submissions = Services::get('submission')->getMany(['contextId' => $journal->getId()]);
+				foreach ($submissions as $submission) {
+					if ($submission->getSubmissionProgress() == 0) { // Not incomplete
+						$this->submissionMetadataChanged($submission);
+						$this->submissionFilesChanged($submission);
```

Comment:
Not sure, but should this only include published submissions? If so, you can add a `status` property to the `getMany` call:

```
Services::get('submission')->getMany(['contextId' => $journal->getId(), 'status' => STATUS_PUBLISHED]);
```

Reply:
Depends on our indexing strategy -- there's value in indexing unpublished content if we don't re-index fully on publication, which may be the case. This needs an audit/clean-up -- I'll do a small assessment as part of this issue and invariably leave some for https://github.com/pkp/pkp-lib/issues/3180 as well.